app/src/main/java/com/cmput301f20t21/bookfriends/ui/add/AddEditViewModel.java

```diff
@@ -72,43 +58,34 @@ public void handleAddBook(
-                            if (newUri != null) {
-                                // image not changed
-                                if (oldUri != null && oldUri.equals(newUri)) {
-                                    successCallback.run(book);
-                                } else { // image is being added or replaced
-                                    String imageName = bookId + "cover";
-                                    // addImage will also replace if file with imageName already exist
-                                    bookRepository.addImage(imageName, newUri).addOnCompleteListener(
-                                            addImageTask -> {
-                                                if (addImageTask.isSuccessful()) {
-                                                    bookRepository.addImageNameToBook(bookId, imageName).addOnCompleteListener(
-                                                            addNameTask -> successCallback.run(book)
-                                                    );
-                                                } else {
-                                                    failCallback.run(BOOK_ERROR.FAIL_TO_ADD_IMAGE);
-                                                }
-                                            }
-                                    );
-                                }
-                            } else if (oldUri != null && newUri == null){
-                                // user deletes image (not yet implemented)
-                            } else { // no image before edit and no image after edit
-                                successCallback.run(book);
-                            }
+        String bookId = oldBook.getId();
+        bookRepository.editBook(bookId, isbn, title, author, description).addOnCompleteListener(
+                editBookTask -> {
+                    if (editBookTask.isSuccessful()) {
+                        Book book = new Book(bookId, isbn, title, author, description, oldBook.getOwner(), BOOK_STATUS.AVAILABLE);
+                        // addImage will also replace if file with imageName already exist
+                        if (newUri != null) {
+                            // when the image is updated
+                            bookRepository.addImage(book.getCoverImageName(), newUri).addOnCompleteListener(
+                                    addImageTask -> {
+                                        if (addImageTask.isSuccessful()) {
+                                            successCallback.run(book);
+                                        } else {
+                                            failCallback.run(BOOK_ERROR.FAIL_TO_ADD_IMAGE);
+                                        }
+                                    }
+                            );
                         } else {
-                            failCallback.run(BOOK_ERROR.FAIL_TO_EDIT_BOOK);
+                            // image is not changed
```

Comment:
it's also possible here that the user deleted the image(we not yet implemented that)

Reply:
can you point out what specifically will break here if they delete the image? I'm assuming that will happen in another vm method deleteImageForBook.

Reply:
I don't think anything will break here, but it's possible that the image fail to delete(although its rare), yeah I agree that can be handled in another method with another async call